### ISSUE TITLE
gcp_cloudfunctions_function is not listing gen2 cloud functions

### ISSUE THREAD

**Issue by vikas-cd** [turbot/steampipe-plugin-gcp#565]
**Describe the bug**
Upon running select * from gcp.gcp_cloudfunctions_function, only gen1 cloud functions are being returned.
![image](https://github.com/turbot/steampipe-plugin-gcp/assets/117085734/8a24dbb3-664a-4e13-8779-d1191f744cd7)

**Steampipe version (`steampipe -v`)**
Example: v0.22.2

**Plugin version (`steampipe plugin list`)**
Example: v0.51.0

**To reproduce**
select * from gcp.gcp_cloudfunctions_function, run this query to check on your side.

**Expected behavior**
Basically all cloud functions should have been listed or picked.

**Additional context**
Gloud CLI output screenshot :
![image](https://github.com/turbot/steampipe-plugin-gcp/assets/117085734/552be1e7-c6ea-47de-89ce-f2877ca48a14)

**Comment by misraved** [comment]
Thanks @vikas-cd for opening this issue 👍!!

As per the discussion, I have assigned this issue to you. Thank you once again for volunteering to fix the code 👍.

Please let us know if you have questions, happy to help!!

**Comment by vikas-cd** [comment]
@misraved can you pls assign this to one of my team member, I am not able to get enough time.
This is his github-id -> ashutoshmore658.
pls allocate this to him.

**Comment by ashutoshmore658** [comment]
@misraved  Hii, I have worked on the similar issue. I will fix this. could you please assign this to me.
